### PR TITLE
docs: add Drag and Drop samples

### PR DESCRIPTION
**Pull request by plagoa**

### CODE REVIEW

**Review by zettca — COMMENTED** [review]
The samples look great 💯

Some notes, most regarding showing the source code - I really think we should bring the code sample to the Storybook, and keep it as simple as we can (focusing on the drag-and-drop)

**Comment by plagoa** [comment]
@zettca all issues have been addressed.

File: docs/guides/dnd/DndKitMultiple/DndKitMultiple.tsx [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+} from "@hitachivantara/uikit-react-core";
+import { Drag } from "@hitachivantara/uikit-react-icons";
+
+import { css } from "@emotion/css";
+
+import { sampleColumns, sampleItems } from "./sampleData";
+import { Column, Item } from "../types";
+import classes from "./styles";
+
+// Fixes a problem we have while dragging node types from the sidebar to the flow in storybook docs mode
```
Comment (MEsteves22) [inline docs/guides/dnd/DndKitMultiple/DndKitMultiple.tsx:32]: We should update the comment since it doesn't apply to this sample

Same for DndKitSingle

**Review by MEsteves22 — APPROVED** [review]
LGTM👌🏻I just left one comment:

File: docs/guides/dnd/Vanilla/Vanilla.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    updatedItems.splice(updatedDroppedIndex + 1, 0, draggedItem as Item);
+
+    setItems(updatedItems);
+  };
+
+  return (
+    <div className={classes.root}>
+      <div className={classes.columnsContainer}>
+        <div className={classes.column}>
+          <div className={classes.columnHeader}>
```
Comment (zettca) [inline docs/guides/dnd/Vanilla/Vanilla.tsx:72]: There's still "dead code" regarding columns in the samples

File: docs/guides/dnd/DndKitMultiple/DndKitMultiple.tsx [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+  const rect = document.getElementById(rootId)?.getBoundingClientRect();
+
+  const docsMode = window.location.search.includes("?viewMode=docs");
+
+  return {
+    ...transform,
+    x: docsMode && rect?.x ? -rect.x + transform.x : transform.x,
+    y: docsMode && rect?.y ? -rect.y + transform.y : transform.y,
+  };
+};
```
[the remark this answers is not in the document]
Comment (plagoa) [reply docs/guides/dnd/DndKitMultiple/DndKitMultiple.tsx:49]: This is needed otherwise it won't work on Storybook in Docs mode.

**Review by zettca — APPROVED** [review]

**Pull request merged**